.github/workflows/e2e.yml

```diff
@@ -38,15 +38,25 @@ jobs:
                   username: ${{ secrets.DOCKERHUB_USERNAME }}
                   password: ${{ secrets.DOCKERHUB_TOKEN }}
 
+            - name: Install Doppler CLI
+              uses: dopplerhq/cli-action@v2
+
             - name: Start docker containers & run cypress
+              env:
+                  DOPPLER_TOKEN: ${{ secrets.DOPPLER_E2E_TOKEN }}
               run: |
                   start_time=$(date -Isecond)
 
                   # start highlight
                   pushd docker;
                   ./start-infra.sh > /tmp/highlight.log 2>&1;
                   docker compose exec -e PSQL_HOST -e PSQL_USER -e PSQL_DB postgres bash -c 'psql -h $PSQL_HOST -U $PSQL_USER $PSQL_DB < /root/init.sql' >> /tmp/highlight.log 2>&1;
-                  ./run-frontend.sh  >> /tmp/highlight.log 2>&1 & ./run-backend.sh >> /tmp/highlight.log 2>&1 &
+                  ./run-backend.sh >> /tmp/highlight.log 2>&1 &
+                  yarn install >> /tmp/highlight.log 2>&1;
+                  doppler run -- yarn build:frontend >> /tmp/highlight.log 2>&1;
+                  yarn workspace @highlight-run/client dev &
+                  yarn workspace highlight.run dev &
+                  yarn workspace @highlight-run/frontend vite preview --port 3000 &
```

Comment:
could also be bundled into a `package.json` `script`, but that might be annoying with the background parallel execution, so this is fine